tests/tests.rs

```diff
@@ -525,6 +525,18 @@ fn test_extension() {
     te.assert_output(&["--extension", ".foo", "a"], "a.foo");
 
     te.assert_output(&["--extension", "foo2"], "one/two/C.Foo2");
+
+    let te2 = TestEnv::new(&[], &["spam.bar.baz", "egg.bar.baz", "yolk.bar.baz.sig"]);
+
+    te2.assert_output(
+        &["--extension", ".bar.baz"],
+        "spam.bar.baz
+        egg.bar.baz",
+    );
+
+    te2.assert_output(&["--extension", "sig"], "yolk.bar.baz.sig");
+
+    te2.assert_output(&["--extension", "bar.baz.sig"], "yolk.bar.baz.sig");
```

Comment:
Could we add a test to check if everything is still Unicode-compatible? Maybe a file with a non-ASCII file-extension.

Reply:
Done with acute `e` and a smiley face.